modules/carousel/Carousel.tsx

```diff
@@ -353,18 +375,156 @@ const Carousel: React.FC<ICarouselProps & WithStylesProps<Theme, ICarouselStyles
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [calculateChildrenSize, alignmentType, scrollableElementRef?.current?.clientWidth]);
 
+  /**
+   * This useEffect makes sure that after some scroll action there's
+   * an item on the center.
+   * This calculates if the scrollLeft where the user stop scrolling
+   * is closer to the next item, or the previous one, and scroll accordingly.
+   */
+  useEffect(() => {
+    if (alignmentType !== "centered" || !autoFocus) {
+      return;
+    }
+
+    const eventListener = () => {
+      if (scrollableElementRef?.current && scrollBaseReference) {
+        const currentScroll = scrollableElementRef.current.scrollLeft;
+        let indexScroll = scrollBaseReference;
+        let indexCount = 0;
+        const childrenSize = calculateChildrenSize();
+
+        if (currentScroll < indexScroll) {
+          // eslint-disable-next-line no-param-reassign
+          scrollableElementRef.current.scrollLeft = indexScroll;
+          setCenteredItemInd(0);
+        }
+
+        const maxPossibleScroll = indexScroll + childrenSize * (items.length - 1);
+        if (currentScroll > maxPossibleScroll) {
+          // eslint-disable-next-line no-param-reassign
+          scrollableElementRef.current.scrollLeft = maxPossibleScroll;
+          setCenteredItemInd(items.length - 1);
+        }
+
+        while (currentScroll >= indexScroll) {
+          indexScroll += childrenSize;
+          indexCount += 1;
+        }
+
+        const highOffset = indexScroll - currentScroll;
+        const lowOffset = currentScroll - (indexScroll - childrenSize);
+        let chosenOffset = indexScroll;
+
+        if (highOffset > lowOffset) {
+          chosenOffset -= childrenSize;
+          indexCount -= 1;
+        }
+
+        if (chosenOffset >= scrollBaseReference && Math.abs(chosenOffset - currentScroll) > 2) {
+          // eslint-disable-next-line no-param-reassign
+          scrollableElementRef.current.scrollLeft = chosenOffset;
+          setCenteredItemInd(indexCount);
+        }
+      }
+    };
+    // eslint-disable-next-line mdx/no-unused-expressions
+    scrollableElementRef?.current?.addEventListener("scroll", debounce(eventListener, 300));
+  }, [
+    alignmentType,
+    autoFocus,
+    calculateChildrenSize,
+    items.length,
+    scrollBaseReference,
+    scrollableElementRef
+  ]);
+
+  /**
+   * This useEffect darkens the elements beside the centered one.
+   * As the user scrolls, this calculates how dark should be based on
+   * the distance from the center to that item.
+   * The furthest from the center, the darker will get
+   */
+  useEffect(() => {
+    if (alignmentType !== "centered" || !scrollableElementRef?.current || !darkenFurthestItems) {
+      return;
+    }
+
+    const element = scrollableElementRef.current;
+
+    const darkenEffectFunc = () => {
```

Comment:
A lot of style properties are being changed one by one in this method. Would it make sense to set `styles` all at once with an object that has all the necessary styles? 

A style object could be built and `item.style` would be set at the end to that object.
What do you think? 💭

Reply:
`item.style` only accepts a string assertion. you can't pass an object directly to `item.style`. 
or were you suggesting a different approach ?

Reply:
I wasn't, no. I was unaware you couldn't assign an object directly. Oh well